http/src/client/mod.rs

```diff
@@ -1462,7 +1463,9 @@ impl Client {
                 Error::CreatingHeader { name, source }
             })?;
 
-            builder = builder.header(AUTHORIZATION, value);
+            builder
+                .headers_mut()
+                .and_then(|headers| headers.insert(AUTHORIZATION, value));
```

Comment:
Can you use an `if let` here and below to be consistent?

Reply:
Done.